test/commands.js

```diff
@@ -437,6 +439,78 @@ let runTransferFromCommand = async (command, state) => {
   return state;
 }
 
+
+//
+// Market Maker commands
+//
+
+let priceFactor = 100000
+
+let getMMMaxClaimableEth = function(state) {
+  if (state.marketMakerMonth >= state.marketMakerPeriods) {
+    help.debug("calculating maxClaimableEth with", state.marketMakerStartingBalance,
+      state.marketMakerClaimedEth,
+      state.returnedWeiForBurnedTokens);
+    return state.marketMakerStartingBalance.
+      minus(state.marketMakerClaimedEth).
+      minus(state.returnedWeiForBurnedTokens);
+  } else {
+    const maxClaimable = state.marketMakerStartingBalance.
+      mul(state.claimablePercentage).dividedBy(priceFactor).
+      mul(state.initialTokenSupply - state.marketMakerBurnedTokens).
+      dividedBy(state.initialTokenSupply).
+      minus(state.marketMakerClaimedEth);
+    return _.max([0, maxClaimable]);
+  }
+}
+
+// TODO: implement finished
+let isMarketMakerFinished = (state) => false
+
+let runMarketMakerSendTokensCommand = async (command, state) => {
+  if (state.marketMaker === undefined) {
+    // doesn't make sense to execute the actual command, let's just assert
+    // that the crowdsale was not funded (in which case there should be MM)
+    assert.equal(false, state.crowdsaleFinalized && state.crowdsaleFunded,
+      "if there's no market Maker, crowdsale should not have been funded");
+  } else {
+    let lifWei = help.lif2LifWei(command.tokens),
+      lifBuyPrice = state.marketMakerBuyPrice.div(priceFactor),
+      tokensCost = new BigNumber(lifWei).mul(lifBuyPrice),
+      fromAddress = accounts[command.from],
+      initialEthBalance = state.ethBalances[command.from] || new BigNumber(0),
+      initialLifBalance = getBalance(state, command.from);
+
+    let shouldThrow = !state.crowdsaleFinalized ||
+      !state.crowdsaleFunded ||
+      state.marketMakerPaused ||
+      (command.tokens == 0) ||
+      isMarketMakerFinished(state);
+
+    try {
+      help.debug('Selling ',command.tokens, ' tokens in exchange of ', web3.fromWei(tokensCost, 'ether'), 'eth');
+      tx1 = await state.token.approve(state.marketMaker.address, lifWei, {from: fromAddress}),
+        tx2 = await state.marketMaker.sendTokens(lifWei, {from: fromAddress}),
+        gas = tx1.receipt.gasUsed + tx2.receipt.gasUsed;
+
+      help.debug("sold tokens to market Maker");
+
+      state.ethBalances[command.from] = initialEthBalance.plus(tokensCost).minus(help.gasPrice.mul(gas));
```

Comment:
why `initialEthBalance` is used here?

Reply:
good question! haha
it sounds misleading but I think it's correct: it's the balance of the `command.from` account before doing the `approve` + `sendTokens`, it's initialized in this function. Maybe it sounded like it's the initial balance when the Market Maker was created or something like that?

Reply:
Yes, that was I understood when I was reading it, we can change it fro `initialSenderWeiBalance` or something like that.